src/node_shapes.jl

```diff
@@ -0,0 +1,227 @@
+"""
+    node_shape(shape::Symbol, clip::Bool; <keyword arguments>)
+
+Create an outline for the graph node. Also, can define a clipping region for the node region.
+
+Returns a dictionary of drawing options and a callable draw function (in this order).
+
+# Arguments
+- `shape::Symbol`: Specify a node shape among `:circle`, `:rectangle` or a `:polygon` (to be implemented).
+- `clip::Bool`: Define a clipping region over the region defined by `shape`.
+
+# Keywords
+- `radius`: Specify if shape is `:circle`.
+- `width`: Specify if shape is `:rectangle`.
+- `height`: Specify if shape is `:rectangle`.
+"""
+function node_shape(shape::Symbol, clip::Bool=false; kwargs...)
+    kw = Dict(collect(kwargs)...)
+    opts = Dict{Symbol, Any}()
+    opts[:shape] = shape
+    opts[:clip] = clip
+    if shape == :rectangle
```

Comment:
currently there is no error message when one writes:
```julia
node_shape(:rectangle, true, radius=12)
```

Maybe have a look at: https://docs.julialang.org/en/v1/manual/types/#%22Value-types%22 to dispatch on the `Symbol`